skiko/src/awtMain/objectiveC/macos/MetalRedrawer.mm

```diff
@@ -121,6 +121,12 @@ JNIEXPORT jlong JNICALL Java_org_jetbrains_skiko_redrawer_MetalRedrawer_createMe
     JNIEnv *env, jobject redrawer, jlong windowPtr, jboolean transparency, jlong adapterPtr, jlong platformInfoPtr)
 {
     @autoreleasepool {
+        NSWindow *window = nil;
+
+        if (windowPtr != 0) {
```

Comment:
`createMetalDevice` requires `window` to always exist. We should write the code with consideration that we always get a window here, without checking it with `if`.

Reply:
I was thinking about putting an assertion in there, do they execute at all?

Reply:
We usually add `check(windowPtr)` on the java side if we are not sure this is always successful.